skrf/time.py

```diff
@@ -127,100 +123,132 @@ def indexes(y, thres=0.3, min_dist=1):
                 rem[sl] = True
                 rem[peak] = False
 
-        peaks = np.arange(y.size)[~rem]
+        peaks = npy.arange(y.size)[~rem]
 
     return peaks
 
-def find_n_peaks(x,n, thres=.9, **kw):
-    '''
-    Find a given number of peaks in a signal
-    '''
-    
-    for dummy  in range(10):
-        
-        idx = indexes(x, **kw)
+def find_n_peaks(x: npy.ndarray, n: int, thres: float = 0.9, **kwargs):
+    """
+    Find a given number of peaks in a signal.
+
+    Parameters
+    ----------
+    x : numpy.ndarray
+        signal
+    n : int
+        number of peaks to search for
+    thres : float, optional
+        threshold, default is 0.9
+    **kwargs : optional keyword arguments passed to :func:`indexes`
+
+    Returns
+    -------
+    peak_idxs : list
+        List containing the numeric indexes of the peaks that were detected
+
+    Raises
+    ------
+    ValueError
+        If no peaks are found.
+    """
+    for dummy in range(10):
+
+        idx = indexes(x, **kwargs)
         if len(idx) < n:
-            thres*=.5
-            
+            thres *= .5
+
         else:
             peak_vals = sorted(x[idx], reverse=True)[:n]
-            peak_idxs =[x.tolist().index(k) for k in peak_vals]
+            peak_idxs = [x.tolist().index(k) for k in peak_vals]
 
             return peak_idxs
-    raise ValueError('Couldnt find %i peaks'%n)
-    
-def detect_span(ntwk):
-    '''
-    detect the correct time-span between two largest peaks
-    '''
+    raise ValueError('Couldnt find %i peaks' % n)
+
+
+def detect_span(ntwk) -> float:
```

Comment:
You can add a type hint for the `ntwk` argument (several occasions in methods of this class):
 `ntwk: Network`

However, this requires you to import the `Network` class at the top: `from .network import Network`

Reply:
I didn't because it created a circular import here... if you have a solution...

Reply:
You're right. I was not aware of that problem, but I found a nice article proposing a solution:
https://adamj.eu/tech/2021/05/13/python-type-hints-how-to-fix-circular-imports/

So, what you can do in `time.py` is a conditional import of `Network` only for the type checking:
```
from __future__ import annotations
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from .network import Network
```
Then, you can do the type hinting for the methods in `time.py`:
```
def detect_span(ntwk: Network) -> float:
```

Reply:
@FranzForstmayr seems to be an expert for type hinting, maybe he can confirm this is a good solution.

Reply:
> I didn't because it created a circular import here... if you have a solution...

using 'Network' works fine. Sphinx even knows how to link to the Network page!

Reply:
Great. No import required at all?

Reply:
hum, flake8 doesn't like finally. OK, I drop it for now.

Reply:
Well, it worked for me in PR #505.

I ended up using a mixture of our tricks above:
```
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from .network import Network
```
But no `from __future__ import annotations`, because it only works for Python >= 3.7. 

So, I had to put `ntwk: 'Network'` as a string.

Reply:
I think that's the best solution right now. When 3.6 support get's dropped in the future we can start to use  `__annotations__` from `futures`. The `TYPE_CHECKING` is a great way to avoid circular imports.